include/reduced_basis/rb_parameters.h

```diff
@@ -300,20 +372,20 @@ class RBParameters
   /**
    * Fill \p param_names with the names of the parameters.
    *
-   * Note: this function was previously deprecated, but now that
-   * multi-step RBParameters objects are possible, this is actually
-   * the most efficient way to get a list of the parameter names
-   * stored on this object.
+   * \deprecated instead of creating a new set of strings here, it's
+   * better to iterate over the RBParameters object directly, using
+   * the iterators from the begin()/end() functions. The .first
+   * will provide the parameter name, .second can be ignored as needed.
```

Comment:
I found 2 legitimate instances in the Akselos code-base where `get_parameter_names()` could not be trivially improved by using iterators instead of making this new object of string copies.

In those cases, it's a simple replacement to iterate yourself over the map, and populate the set with the results.

I could go either way. I think this function has some limited usefulness, but mostly I want to discourage people from using it unnecessarily.

Reply:
I've removed the deprecation comment.

I've also changed the function so it just returns a std::set with copy-elision, rather than taking it as a parameter and clearing it.